src/os/bluestore/BlueStore.h

```diff
@@ -1740,7 +1738,7 @@ class BlueStore : public ObjectStore,
 	// may become true outside qlock, and we need to make
 	// sure those threads see waiters and signal qcond.
 	++kv_submitted_waiters;
-	if (_is_all_kv_submitted()) {
+	if (!q.empty() && _is_all_kv_submitted()) {
```

Comment:
what's about q.empty() == true case

Reply:
Move q.empty() from _is_all_kv_submitted to this.  Most call _is_all_kv_submitted is the situation q.empyt() == false.  So i move code .  Because flush() should first check q whether empty.

Reply:
Before your changes q.empty() == true caused return here. Now it doesn't. Hence the new behavior isn't identical. Is that expected? I suppose proper condition should be (q.empty() || is_all_kv_submitted())

Reply:
Yes. you are right. Thanks!